src/app/ngrx/reducers/episode-metrics.reducer.ts

```diff
@@ -122,25 +107,26 @@ export function EpisodeMetricsReducer(state: EpisodeMetricsModel[] = initialStat
     // so am handling the ROUTER_NAVIGATION event here to catch changes to charted episodes via the route
     case ACTIONS.ActionTypes.CUSTOM_ROUTER_NAVIGATION: {
       const payload: ACTIONS.CustomRouterNavigationPayload = action['payload'];
-      let newState: EpisodeMetricsModel[];
-      if (payload.routerState.episodeIds && payload.routerState.episodeIds.length > 0) {
-        newState = [...state];
-        const {podcastSeriesId, episodeIds} = payload.routerState;
-        let episode: EpisodeMetricsModel;
-        episodeIds.forEach(id => {
-          const epIdx = episodeIndex(newState, id, podcastSeriesId);
-          if (epIdx > -1) {
-            episode = {id, seriesId: podcastSeriesId, ...newState[epIdx], charted: true};
-            newState = [...newState.slice(0, epIdx), episode, ...newState.slice(epIdx + 1)];
-          } else {
-            episode = {id, seriesId: podcastSeriesId, charted: true};
-            newState = [episode, ...newState];
-          }
-        });
-        return newState;
+      // update existing entries
```

Comment:
at least we're now a step closer to moving this state out of here and only keeping it on the route